### PR TITLE
ADCM-1467 Collect logs for timed-out jobs

### PR DESCRIPTION
**Pull request by dgusakov**

### CODE REVIEW

**Review by skhomuti — APPROVED** [review]

File: adcm_client/objects.py [hunk 1]
```diff
@@ -754,8 +758,9 @@ def try_wait(self, timeout=None):
 
         return status
 
-    def _log_jobs(self, status):
-        for job in self.job_list(status=status):
+    def _log_jobs(self, status=None):
+        job_list = self.job_list(status=status) if status else self.job_list()
+        for job in job_list:
```
Comment (acmnu) [inline adcm_client/objects.py:763]:
```suggestion
    def _log_jobs(self, **filters):
      for job in self.job_list(**filters):
```

There is no real needs to pass status when you can pass any of job ojbect attribute. 

But in that case you have to call this function as follows:

```python
            if log_failed and status == "failed":
                self._log_jobs(status="failed")
```

Or even better:
```python
            if log_failed and status == "failed":
                self._log_jobs(status=status)
```

File: adcm_client/objects.py [hunk 2]
```diff
@@ -738,11 +738,15 @@ def job_list(self, paging=None, **args) -> "JobList":
...
-        if log_failed and status == "failed":
-            self._log_jobs('failed')
+        try:
+            status = self.wait_for_attr("status",
+                                        self._END_STATUSES,
+                                        timeout=timeout)
+            if log_failed and status == "failed":
+                self._log_jobs('failed')
+        except TimeoutError as e:
+            self._log_jobs()
```
Comment (acmnu) [inline adcm_client/objects.py:748]: If you guard line 746 with if, why you do nothing with line 748? Should log_failed also control timeout?

**Review by acmnu — CHANGES_REQUESTED** [review]

File: adcm_client/objects.py (hunk 1 above)
Comment (dgusakov) [reply adcm_client/objects.py]: Fixed

File: adcm_client/objects.py (hunk 2 above)
Comment (dgusakov) [reply adcm_client/objects.py]: Yes it does. Fixed.

**Review by acmnu — APPROVED** [review]

**Pull request merged**